core/src/impl/Kokkos_Profiling_Interface.cpp

```diff
@@ -52,6 +52,20 @@ namespace Kokkos {
        	return (NULL != initProfileLibrary);
     }
 
+    void pushRegion(const std::string& kName) {
+	if( NULL != pushRegionCallee ) {
+	    Kokkos::fence();
```

Comment:
What do you mean with execution space instance? This is called from the master thread not from any instance.